src/main/windows/html_email.ts

```diff
@@ -0,0 +1,299 @@
+    subject,
+    from,
+    networkButtonLabelText: tx('load_remote_content'),
+  }))
+
+  nativeTheme.on('updated', () => {
+    window.webContents.ipc.emit('theme-update')
+  })
+
+  window.webContents.ipc.handle(
+    'get-theme',
+    async () => (await loadTheme(DesktopSettings.state.activeTheme)).data
+  )
+
+  window.loadFile(
+    join(
+      htmlDistDir(),
+      'electron_html_email_view/electron_html_email_view.html'
+    )
+  )
+
+  window.webContents.on('will-navigate', (e: electron.Event, _url: string) => {
+    // Prevent drag-and-drop from navigating the Electron window, which can happen
+    // before our drag-and-drop handlers have been initialized.
+    e.preventDefault()
+  })
+
+  window.once('ready-to-show', () => {
+    window.show()
+  })
+
+  window.once('focus', () => {
+    window.webContents.setZoomFactor(DesktopSettings.state.zoomFactor)
+  })
+  window.on('close', () => {
+    context_menu_handle?.()
+    delete open_windows[window_id]
+  })
+
+  let sandboxedView: BrowserView = makeBrowserView(false, htmlEmail)
+  window.setBrowserView(sandboxedView)
+  let context_menu_handle = createContextMenu(window, sandboxedView.webContents)
+
+  window.webContents.ipc.handle(
+    'html-view:resize-content',
+    (_ev, bounds: Electron.Rectangle) => {
+      const window_bounds = window.getBounds()
+      const new_y = Math.floor(window_bounds.height - bounds.height)
+      sandboxedView?.setBounds({
+        ...bounds,
+        y: new_y,
+      })
+    }
+  )
+
+  window.webContents.ipc.handle(
+    'html-view:change-network',
+    async (_ev, allow_network: boolean) => {
+      if (
+        allow_network &&
+        DesktopSettings.state.HTMLEmailAskForRemoteLoadingConfirmation
+      ) {
+        const result = await dialog.showMessageBox(window, {
+          message: tx('load_remote_content_ask'),
+          checkboxLabel: tx('do_not_ask_again'),
+          buttons: [tx('no'), tx('yes')],
+          type: 'none',
+          icon: '',
+          defaultId: 0,
+          cancelId: 0,
+        })
+        if (result.response === 0) {
+          throw new Error('user denied')
+        }
+        if (result.checkboxChecked) {
+          DesktopSettings.update({
+            HTMLEmailAskForRemoteLoadingConfirmation: false,
+          })
+        }
+      }
+
+      const bounds = sandboxedView?.getBounds()
+      window.removeBrowserView(sandboxedView)
+      context_menu_handle()
+      sandboxedView.webContents.close()
+      sandboxedView = makeBrowserView(allow_network, htmlEmail)
+      window.setBrowserView(sandboxedView)
+      context_menu_handle = createContextMenu(window, sandboxedView.webContents)
+      if (bounds) sandboxedView.setBounds(bounds)
+
+      // for debugging email
+      // sandboxedView.webContents.openDevTools({ mode: 'detach' })
+    }
+  )
+
+  // for debugging wrapper
+  // window.webContents.openDevTools({ mode: 'detach' })
+}
+
+const CSP_DENY =
```

Comment:
Why do you need `blob:` without enabling JavaScript? What do you think about this?
* `CSP_DENY = "default-src 'none'; font-src 'self' data:; frame-src 'self' data:; img-src 'self' data:; media-src 'self' data:; style-src 'self' data: 'unsafe-inline'; form-action 'none'; navigate-to 'none';"`
* `CSP_ALLOW = "default-src 'none'; font-src 'self' data: http: https:; frame-src 'self' data: http: https:; img-src 'self' data: http: https:; media-src 'self' data: http: https:; style-src 'self' data: 'unsafe-inline' http: https:; form-action 'none'; navigate-to 'none';`

Reply:
good catch, yeah we probably don't need it the `blob:` uri. also we don't want scripts in emails at all.

Reply:
- `style-src` not many email clients support remote styles, even support for the style tag is not high, I saw many recommendations of inline styles so I wouldn't allow them.
- `frame-src` - frames open security holes, imagine some kind of nested frame trickery that has javascript enabled again (I won't trust chromium here, also frames in emails sounds like a supper bad idea in general)
- `navigate-to` <img width="804" alt="csp checker site says it is an invalid option" src="https://user-images.githubusercontent.com/18725968/230657419-9c1a07aa-6501-47ff-a8c6-5a343b701ffe.png">